airflow/models/dagrun.py

```diff
@@ -313,14 +360,16 @@ def update_state(self, session: Session = None) -> List[TI]:
         :rtype ready_tis: list[airflow.models.TaskInstance]
         """
 
+        start_dttm = timezone.utcnow()
```

Comment:
Should we change this method to also update state of the tasks? I think it is only used in two places (schedule_job and backfill_job) and the "update_state" method returning tasks sounds a bit weird. I'd rather think about update the task instances as well when update_state is run so that it is all encapsulated within one "update_state" method.

Reply:
Already had that thought :) 
https://github.com/apache/airflow/blob/b28d67332bb052441bc2ffd00030d620aac381c8/airflow/jobs/scheduler_job.py#L1565-L1567

Reply:
Any further thoughts ?

Reply:
Defer to later I think -- not changed in name or behaviour in this PR.